lablib/lib/imageio.py

```diff
@@ -23,72 +23,47 @@
     "display_height": 1080,
 }
 
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
 
-class ImageIOBase:
-    log = logging.getLogger(__name__)
-    log.setLevel(logging.DEBUG)
 
-    def __init__(self, *args, **kwargs):
-        for k, v in kwargs.items():
-            setattr(self, k, v)
-        self.update(*args, **kwargs)
-
-    def __getitem__(self, k: str) -> Any:
-        return getattr(self, k)
-
-    def __setitem__(self, k: str, v: Any) -> None:
-        if hasattr(self, k):
-            setattr(self, k, v)
-        else:
-            raise AttributeError(f"Attribute is not implemented: {k}")
-
-    def __repr__(self) -> str:
-        return f"{self.__class__.__name__}{self.__dict__}"
-
-    def update(self, **kwargs) -> None:
-        """Update operator attributes by calling implemented setter of keyword argument key."""
-        self.log.warning(f"Update not implemented for {self.__class__.__name__}")
-        pass
-
-    @property
-    def path(self) -> Path:
-        return self._path
-
-    @path.setter
-    def path(self, path: Union[str, Path]) -> None:
-        if isinstance(path, str):
-            path = Path(path)
-        if not path.exists():
-            raise Exception(f"Path does not exist: {path}")
+@dataclass
+class ImageInfo:
+    """ImageInfo class for reading image metadata."""
 
-        self._path = path
+    path: Path = field(default_factory=Path)
 
-    @property
-    def filepath(self) -> Path:
-        return self.path
+    # fmt: off
+    width: int = field(default=IMAGE_INFO_DEFAULTS["width"], init=False, repr=True)
```

Comment:
Is there any particular reason the default values from Constant should not be part of initialization of the class? 
`init=False` should not be rather `init=True`?

Reply:
yeah i put these as `False` since i don't want a user to be able to manually set these values when creating the instance as they will be populated from `iinfo`

Reply:
the only field i really want users to be able to set is `path`. all the other values will be populated from `iinfo` as mentioned before